From c3ba4c57f2b1e2c82c565da2590d6a120fd2c32a Mon Sep 17 00:00:00 2001
From: Craig Kewley <craigx.kewley@intel.com>
Date: Fri, 18 Aug 2017 09:33:42 +0100
Subject: [PATCH 059/100] Revert "Revert "ASoC: Intel: Boards: add param to set
 codecs to dummy""

This reverts commit 9711c7315edb868ad3296a42fea70363acbc0f74.
---
 sound/soc/intel/boards/bxt_gpmrb.c | 20 ++++++++++++++++++++
 1 file changed, 20 insertions(+)

diff --git a/sound/soc/intel/boards/bxt_gpmrb.c b/sound/soc/intel/boards/bxt_gpmrb.c
index 83fbd94..863f1e9 100644
--- a/sound/soc/intel/boards/bxt_gpmrb.c
+++ b/sound/soc/intel/boards/bxt_gpmrb.c
@@ -30,6 +30,10 @@
 #define CHANNELS_QUAD 4
 #define CHANNELS_EIGHT 8
 
+static int dummy_codecs;
+module_param(dummy_codecs, int, 0444);
+MODULE_PARM_DESC(dummy_codecs, "Set all DAI link codecs to dummy");
+
 static struct snd_soc_dai_link broxton_gpmrb_dais[];
 
 enum {
@@ -527,9 +531,25 @@ static int broxton_gpmrb_tdf8532_startup(struct snd_pcm_substream *substream)
 	.fully_routed = true,
 };
 
+static void broxton_set_dummy_codecs(void)
+{
+	int i;
+
+	for (i = 0; i < broxton_gpmrb.num_links; i++) {
+		broxton_gpmrb.dai_link[i].codec_name = "snd-soc-dummy";
+		broxton_gpmrb.dai_link[i].codec_dai_name = "snd-soc-dummy-dai";
+	}
+
+	dev_info(broxton_gpmrb.dev, "Codecs set to dummy\n");
+}
+
 static int broxton_audio_probe(struct platform_device *pdev)
 {
 	broxton_gpmrb.dev = &pdev->dev;
+
+	if (dummy_codecs)
+		broxton_set_dummy_codecs();
+
 	return snd_soc_register_card(&broxton_gpmrb);
 }
 
-- 
1.9.1

